feat: update calorie counter project

scissorsneedfoodtoo opened the pull request:
Checklist:

<!-- Please follow this checklist and put an x in each of the boxes, like this: [x]. It will ensure that our team takes your pull request seriously. -->

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a [descriptive title](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request?id=prepare-a-good-pr-title) (**not** a vague title like `Update index.md`)

<!-- Feel free to add any additional description of changes below this line -->

Link to project on CodeSandbox: https://qdvszy.csb.app/

List of concepts covered: 

- [x] `getElementById`
- [x] function declarations and function calls
- [x] regex global flag
- [x] regex +-,s
- [x] `replace()`
- [x] `match` method
- [x]  regex i global flag
- [x] `querySelector` and `querySelectorAll`
- [x] template strings
- [x] `innerHTML`
- [x] `.insertAdjacentHTML()`
- [x] ` e.preventDefault();`
- [x] if/else
- [x] `classList`
- [x] `remove` method
- [x] `const` and `let`
- [x] `alert`
- [x] booleans
- [x] `Array.from`
- [x] for loop
- [x] `add` method
- [x] `addEventListener`

jdwilkin4 reviewed (commented):
I think it looks good.  👍 
Just left a few comments

jdwilkin4 commented:
@scissorsneedfoodtoo 

I created a list of the concepts covered 👍

scissorsneedfoodtoo commented:
@jdwilkin4, wow, thank you for all of your help creating that list of topics covered! I was finally going to start on that, but just saw that you already added one 👍

scissorsneedfoodtoo commented on intermediate-javascript-calorie-counter/index.js:
```diff
@@ -1,84 +1,126 @@
+    - Explain that number inputs only allow e between digits. Match e between specific digits: /1e3/i
+    - Explain character classes to search for digits 0-9. Refactor to /[0-9]e[0-9]/i
+    - Explain the + character to match 1 or more of whatever precedes it: /[0-9]+e[0-9]+/i
+    - Explain that \d matches digits, and can replace [0-9]. Refactor to /\d+ed+/i
```
Awesome, I think so, too. Between this and the other function, it should be enough regex for the required Palindrome Checker project.

scissorsneedfoodtoo commented on intermediate-javascript-calorie-counter/index.js:
```diff
@@ -1,84 +1,126 @@
+    Could teach the .split() and .join()
+    methods to do this with a for loop, then refactor
+    to use regex. Might also be able to have learners
+    practice with the .reverse() method here just for fun.
```
I agree that covering all this in one function might be a bit much, and possibly discouraging if we do a bunch of refactoring into regex.

But I was looking at some common solutions for the Palindrome Checker project, and it's pretty likely that people will try to use the `.split()`, `.reverse()`, and `.join()` methods. 

It's a bit unnatural to use those just to check for a few characters, but I thought it might be a decent way to cover those concepts before learners need it.

I'll think more about this. There may be a good way to use `.split()`, a `for` loop, and `.match()` or `.test()` to do simple matches on an array of strings, then to get more fancy with a single regular expression, the global flag, and `.replace()`.

scissorsneedfoodtoo commented on intermediate-javascript-calorie-counter/index.js:
```diff
@@ -1,84 +1,126 @@
+const addEntryButton = document.getElementById('add-entry');
+const clearButton = document.getElementById('clear');
+const output = document.getElementById('output');
+let error = false;
```
Thanks for catching that. `isError` is definitely a better name here. I'll make sure to change that.

scissorsneedfoodtoo commented:
Hey @jdwilkin4, thanks again for your previous review.

I believe I addressed most of the comments, and either fixed the issue or left a comment that will help later when breaking down the project.

The only thing I wasn't able to address was the bit about regex and possibly doing some refactoring. Looking at the projects that come before the first Palindrome Checker project, this might be the only chance we have to teach some basic regex and methods like `.split()`, `.reverse()`, and `.join()`.

But I was able to simplify the HTML and CSS quite a bit, so that should give us more steps to do that sort of refactoring in the project if we decide to go that route.

Could you take another look when you get a chance and let me know if you notice anything else?

jdwilkin4 reviewed (approved):
I playing around the codesandbox and I think this is ready to get merged.

I did notice two things which could be opportunities for error steps and have campers put in a fix. 

No.1:
The entry start counting at 0 but I think it would be nice if it started at 1.
<img width="204" alt="Screen Shot 2022-10-09 at 8 45 05 PM" src="https://user-images.githubusercontent.com/67210629/194796588-01f3f9c3-a4ae-4068-bfe9-483221c96df0.png">

We could point that out to campers and have then fix it in a step.

No.2:
Even though we are going to be providing the number of calories to campers it is still possible to have negative calories.

<img width="391" alt="Screen Shot 2022-10-09 at 8 46 15 PM" src="https://user-images.githubusercontent.com/67210629/194796653-0b50b351-cd7d-4c4d-8224-6479d2278ecc.png">

Maybe we could have campers add in an extra condition that shows a message if there is a possibility of negative calorie count.

scissorsneedfoodtoo commented:
Awesome, thanks for pointing both of those things out. They should be easy fixes or additions.

I'll go ahead and add those, merge this, and will add it to the spreadsheet.

scissorsneedfoodtoo commented:
Okay, this should be ready to merge.

Adding an entry for a meal or exercise now starts at 1 instead or 0:

![image](https://user-images.githubusercontent.com/2051070/198523730-e702fe52-4dc5-4f7a-bbeb-39d9ecc7836b.png)

And there's some better output reporting with basic colors and other metrics:

![image](https://user-images.githubusercontent.com/2051070/198523841-96530fd7-7ac1-41c8-ac8f-10d54dc35591.png)

![image](https://user-images.githubusercontent.com/2051070/198523986-dda8d757-f947-4037-9b78-18eb37201003.png)
